debezium-connector-sqlserver/src/test/java/io/debezium/connector/sqlserver/IncrementalSnapshotCollationSortOrderMismatchIT.java

```diff
@@ -0,0 +1,395 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.sqlserver;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import io.debezium.config.CommonConnectorConfig;
+import io.debezium.config.Configuration;
+import io.debezium.connector.sqlserver.util.TestHelper;
+import io.debezium.jdbc.JdbcConnection;
+import io.debezium.pipeline.source.snapshot.incremental.AbstractSnapshotTest;
+import io.debezium.util.Testing;
+
+/**
+ * Tests for <a href="https://issues.redhat.com/browse/DBZ-7359">DBZ-7359</a>
+ */
+public class IncrementalSnapshotCollationSortOrderMismatchIT extends AbstractSnapshotTest<SqlServerConnector> {
```

Comment:
I don't think it is necessary to inherit from `AbstractSnapshotTest` and fit into its structure. You can make a completely standalone test out of it if it would be easier.

Reply:
I tried that initially but the test leverages these 3 methods:
 AbstractConnectorTest.initializeConnectorTestFramework() 
 AbstractSnapshotTest.startConnector()
 AbstractSnapshotTest.sendAdHocSnapshotSignal() 
and I wanted to avoid cutting and pasting the code into it. Let me know if you'd like me to do this and make it completely standalone.